src/dataaggregator/aggregator/metricPoller.go

```diff
@@ -101,22 +109,23 @@ func (m *MetricPoller) doAggregate(appId string, metricType string, metrics []*m
 			sum += intValue
 		}
 	}
-
+	var avgAppMetric *appmetric.AppMetric
 	if count == 0 {
-		return &appmetric.AppMetric{
+		avgAppMetric = &appmetric.AppMetric{
```

Comment:
@fraenkel When there is no metric from metric-collector, the aggregator will construct an zero-value AppMetric instance and persist it.